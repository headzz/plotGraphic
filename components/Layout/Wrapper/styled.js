import styled from "styled-components";

export const WrapperStyled = styled.div`
  padding: 2rem 4rem;
`;
