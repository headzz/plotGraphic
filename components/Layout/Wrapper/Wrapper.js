import PropTypes from "prop-types";
import React from "react";
import { WrapperStyled } from "./styled";

/**
 * @component
 * @name Wrapper
 * @description Defines the container for the layout.
 * @namespace components components/Wrapper
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */
const Wrapper = ({ children }) => {
  return <WrapperStyled>{children}</WrapperStyled>;
};

export default Wrapper;

Wrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
