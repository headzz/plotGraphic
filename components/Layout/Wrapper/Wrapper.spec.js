import React from "react";
import { render, cleanup } from "@testing-library/react";
import Wrapper from "./Wrapper";

afterEach(cleanup);

test("render layout Wrapper", () => {
  const { container } = render(
    <Wrapper>
      <div>1</div>
    </Wrapper>
  );
  expect(container.firstChild).toHaveStyle("padding: 2rem 4rem");
  expect(container.firstChild).toMatchSnapshot();
});
