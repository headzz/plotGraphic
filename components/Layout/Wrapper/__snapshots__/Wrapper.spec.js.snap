// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render layout Wrapper 1`] = `
<div
  class="sc-bdVaJa bkKsTH"
>
  <div>
    1
  </div>
</div>
`;
